### DISCUSSION TITLE
Convert PNG to ICO

### DISCUSSION THREAD

**Discussion by katastrophe92** [ImageMagick/ImageMagick#6248] [Help]
### ImageMagick version

7.1.1-6

### Operating system, version and so on

Windows 11 pro last

### Description

Hello ImageMagick community! I have just a quick question: can this software convert .png to .ico? Hope someone can help me. Thanks!

**Comment by snibgo** [comment]
Yes. For example, if in.png has 256x256 pixels:
```
magick in.png -define icon:auto-resize out.ico
```

**Comment by GeeMack** [comment]
You can convert a PNG image into a multi-size icon image with a command like this...

    magick image.png -define icon:auto-resize="256,128,64,32,16" result.ico

The output is an icon file containing images in each of the specified dimensions.

**Comment by katastrophe92** [comment]
@GeeMack and @snibgo thanks for your answers ☺. Do I need to type the image title one by one or it is possible to convert a folder content or multiple .png files? If in batch is possible what do I need to type? Thanks 👍

**Comment by snibgo** [comment]
In the directory that contains your input files, which I assume are `*.png`:
```
magick mogrify -define icon:auto-resize="256,100" -format ico *.png
```
In the quotes put whatever sizes you want, or omit `="256,100"` to get default sizes.
